security: add more examples (#6645)
tldr-pages/tldr: pages/osx/security.md

- Create a keychain:

`security create-keychain -p {{password}} {{keychain.name}}`

> - Set a certificate to use with a website or [s]ervice by its [c]ommon name (this will fail if several certificates with the same common name exist):

Reviewer: ```suggestion
- Set a certificate to use with a website or service by its common name (fails if several certificates with the same common name exist):
```

Author: Please see my comment https://github.com/tldr-pages/tldr/pull/6645/files#r720561999 , I'm unsure based on other pages if this is part of the repository's convention

Author: Implemented the "this will fail" -> "fails" change

> `security set-identity-preference -s {{URL|hostname|service}} -c {{friendly_certificate_name_in_keychain}} {{path/to/file.keychain}}`

Reviewer: I'd put the common name in quotes, because it often contains spaces.

```suggestion
`security set-identity-preference -s {{URL|hostname|service}} -c "{{common_name}}" {{path/to/file.keychain}}`
```

> `security set-identity-preference -s {{URL | hostname | service}} -c {{friendly_certificate_name_in_keychain}} {{path/to/file.keychain}}`

Reviewer 2: ```suggestion
`security set-identity-preference -s {{URL|hostname|service}} -c {{friendly_certificate_name_in_keychain}} {{path/to/file.keychain}}`
```

The spaces don't fit our syntax

> - Set a preferred certificate to use with a website or [s]ervice by its [c]ommon name (note that this will fail if several certificates with the same common name exist):
> 
> `security set-identity-preference -s {{URL, hostname, or service}} -c {{friendly certificate name in keychain}} {{path such as /Library/Keychains/System.keychain}}`

Reviewer 3: ```suggestion
- Set a certificate to use with a website or [s]ervice by its [c]ommon name (this will fail if several certificates with the same common name exist):

`security set-identity-preference -s {{URL | hostname | service}} -c {{friendly_certificate_name_in_keychain}} {{path/to/file.keychain}}`
```

> - Add a certificate from file to a [k]eychain (if -k isn't specified, the default keychain is used):

Reviewer: I don't think, we ever used brackets for indicating this.

```suggestion
- Add a certificate from file to a keychain (if -k isn't specified, the default keychain is used):
```

Author: I've observed the [flag_letter] syntax in a number of other pages, so I was following that convention.
e.g.
- /pages/linux/xsel.md:- Use a command's output as input of the **clip[b]oard** (equivalent to `Ctrl + C`):
- /pages/common/chmod.md:- Give **[o]thers** (not in the file owner's group) the same rights as the **[g]roup**:

The output of `grep -r -E '\[[a-zA-Z]\]' . | grep pages/` in the repository root shows this is quite a ubiquitous convention.

Reviewer 2: We may want to revert this commit. You observed correctly.

Reviewer 2: Oh you never added it.. Well that makes it a little easier :smile:

> - Add a certificate from file to a [k]eychain (if -k not specified the default keychain is used):

Reviewer 2: ```suggestion
- Add a certificate from file to a [k]eychain (if -k isn't specified, the default keychain is used):
```
Small grammatical error.

> `security add-certificates -k {{keychain.name}} {{path/to/file}}`

Reviewer: Aren't certs usually `.pam` files? I'd do it like this:

```suggestion
`security add-certificates -k {{keychain.name}} {{path/to/file.pam}}`
```

Author: They are usually .p**e**m, .p**a**m is for sshd

Author: I've added the .pem extension

> `security add-certificates [-k {{keychain.name}}] {{path}}`

Reviewer 3: ```suggestion
`security add-certificates -k {{keychain.name}} {{path/to/file}}`
```
